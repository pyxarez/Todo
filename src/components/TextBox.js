import React, { Component } from 'react';

import './TextBox.res/style.css';

export default class TextBox extends Component {
  static propTypes = {
    isFocused: React.PropTypes.bool,
    onClick: React.PropTypes.func.isRequired,
    placeholder: React.PropTypes.string
  }
  
  static defaultProps = {
    isFocused: false
  }

  handleClick = () => {
    const value = this.textInput.value;
    if (value.trim() === '') {
      alert("Type something please");
      return;
    }
    
    this.textInput.value = '';
    this.props.onClick(value);
  }

  handleKeyPress = (e) => {
    if (e.key === "Enter") this.handleClick();
  }

  render() {
    const {
      isFocused,
      placeholder,
    }= this.props;

    return (
      <div className="my-textbox-component">
        <input
          autoFocus={isFocused}
          onKeyPress={this.handleKeyPress}
          ref={(input) => { this.textInput = input }}
          className="textbox-input"
          type="text"
          placeholder={placeholder}/>
        <button onClick={this.handleClick}>Add</button>
      </div>
    );
  }
}
