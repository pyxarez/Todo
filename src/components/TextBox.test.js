import React from 'react';
import { mount } from 'enzyme';

import TextBox from './TextBox';

const setup = () => {
  const props = {
    isFocused: true,
    placeholder: 'Kaaarl',
    onClick: jest.fn()
  }
  const wrapper = mount(
    <TextBox {...props} />
  );

  return {
    props,
    wrapper
  }
}

describe('Components :: TextBox', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick prop through the handleKeyPress call', () => {
    const { wrapper, props } = setup();
    const value = 'lol';

    const input = wrapper.find('input');
    input.node.value = value;
    input.simulate('keypress', {
      key: 'Enter'
    });

    expect(props.onClick)
      .toBeCalledWith(value);
  });

  it('should clear input after successful onClick call', () => {
    const { wrapper, props } = setup();
    const value = 'lol';

    const input = wrapper.find('input');
    input.node.value = value;
 
    const button = wrapper.find('button');
    button.simulate('click');
    
    expect(input.node.value).toBe('');
  });

  it('should call handleKeyPress with non-enter key', () => {
    const { wrapper, props } = setup();

    const input = wrapper.find('input');
    input.simulate('keypress', {
      key: 'e'
    });

    expect(props.onClick).not.toBeCalled();
  });

  it('should call onClick prop through the handleClick call,', () => {
    const { wrapper, props } = setup();
    const value = 'Vassa';

    const input = wrapper.find('input');
    input.node.value = value;

    const button = wrapper.find('button');
    button.simulate('click');

    expect(props.onClick).toBeCalledWith(value);
  });

  it('should call handleClick with empty input', () => {
    const { wrapper, props } = setup();
    window.alert = jest.fn();
    
    const button = wrapper.find('button');
    button.simulate('click');

    expect(window.alert).toBeCalled();
  });
});
