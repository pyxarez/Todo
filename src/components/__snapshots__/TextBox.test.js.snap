exports[`Components :: TextBox should render self 1`] = `
<TextBox
  isFocused={true}
  onClick={[Function]}
  placeholder="Kaaarl">
  <div
    className="my-textbox-component">
    <input
      autoFocus={true}
      className="textbox-input"
      onKeyPress={[Function]}
      placeholder="Kaaarl"
      type="text" />
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
</TextBox>
`;
